lib/addons/flagger/index.ts

```diff
@@ -0,0 +1,80 @@
+import { HelmAddOn, HelmAddOnUserProps, HelmAddOnProps } from "../helm-addon";
+import { Construct } from 'constructs';
+import { Values, ClusterInfo } from "../../spi";
+import merge from "ts-deepmerge";
+
+/**
+ * User provided options for the FlaggerAddonProps values.
+ */
+export interface FlaggerAddOnProps extends HelmAddOnUserProps {
+  
```

Comment:
public docs for public members of the interface.

Reply:
I am unsure what you are asking for by that.

Reply:
Docs for public members of the interface. Jsdocs.

Reply:
I don't believe I've done a jsdoc yet, I'll have to look into it.

Reply:
`/**
 * User provided options for the FlaggerAddonProps values.
 * @extends {HelmAddOnUserProps} is needed to set Helm chart values for the interface.
 * @prop {boolean} "determines if prometheus is installed with the addon."
 * @prop {MeshProviderOptions} "Determines what value for mesh provider is used."
 */
export interface FlaggerAddOnProps extends HelmAddOnUserProps {

  installPrometheus?: boolean;
  meshProvider?: MeshProviderOptions;
}`

Reply:
Just regular jsdocs on the public member. That means doc above `installPrometheus` and doc above `meshProvider` so that when developers use your addon, they will see what it means.

Reply:
so more like this?

 /**
   * installPrometheus controls if prometheus is installed with the addon or not.
   */
  installPrometheus?: boolean;
  /**
   * meshProvider controls what mesh provider from MeshProviderOptions enums list is used.
   */
  meshProvider?: MeshProviderOptions;

Reply:
Please don't repeat the name of the field in the doc. 
```
/**
 * Controls if prometheus is installed with the addon or not.
 * default: true
 */
installPrometheus?: boolean;
/*
```

Reply:
That makes sense, so more like this:
/**
   * Controls if prometheus is installed with the addon or not.
   * default: true
   */
  installPrometheus?: boolean;
  /**
   * Controls what mesh provider from MeshProviderOptions enums list is used.
   * default: MeshProviderOptions.KUBERNETES
   */
  meshProvider?: MeshProviderOptions;

Reply:
Yes. Note the formatting of comments.